java/src/main/java/io/cucumber/cucumberexpressions/CucumberExpression.java

```diff
@@ -61,9 +81,18 @@ private String rewriteToRegex(Node node) {
     }
 
     private static String escapeRegex(String text) {
-        return ESCAPE_PATTERN.matcher(text).replaceAll("\\\\$0");
+        int length = text.length();
```

Comment:
Actually kinda neat. Perhaps this method could be extracted to it's own package private class.

Reply:
Thanks. Would it make sense to move it to https://github.com/cucumber/gherkin/blob/main/java/src/main/java/io/cucumber/gherkin/StringUtils.java ?

Reply:
I would prefer a dedicated class just for this method. The static block shouldn't stray too far from the loop that uses it.